cmd/soroban-rpc/internal/methods/get_ledger_entries.go

```diff
@@ -101,32 +113,58 @@ func NewGetLedgerEntriesHandler(logger *log.Entry, ledgerEntryReader db.LedgerEn
 		}
 
 		for _, ledgerKeyAndEntry := range ledgerKeysAndEntries {
-			keyXDR, err := xdr.MarshalBase64(ledgerKeyAndEntry.Key)
-			if err != nil {
-				logger.WithError(err).WithField("request", request).
-					Infof("could not serialize ledger key %v", ledgerKeyAndEntry.Key)
-				return GetLedgerEntriesResponse{}, &jrpc2.Error{
-					Code:    jrpc2.InternalError,
-					Message: fmt.Sprintf("could not serialize ledger key %v", ledgerKeyAndEntry.Key),
+			switch request.Format {
+			case xdr2json.FormatJSON:
+				keyJs, err := xdr2json.ConvertInterface(ledgerKeyAndEntry.Key)
+				if err != nil {
+					return GetLedgerEntriesResponse{}, &jrpc2.Error{
+						Code:    jrpc2.InternalError,
+						Message: err.Error(),
+					}
+				}
+				entryJs, err := xdr2json.ConvertInterface(ledgerKeyAndEntry.Entry.Data)
+				if err != nil {
+					return GetLedgerEntriesResponse{}, &jrpc2.Error{
+						Code:    jrpc2.InternalError,
+						Message: err.Error(),
+					}
 				}
-			}
 
-			entryXDR, err := xdr.MarshalBase64(ledgerKeyAndEntry.Entry.Data)
-			if err != nil {
-				logger.WithError(err).WithField("request", request).
-					Infof("could not serialize ledger entry data for ledger entry %v", ledgerKeyAndEntry.Entry)
-				return GetLedgerEntriesResponse{}, &jrpc2.Error{
-					Code:    jrpc2.InternalError,
-					Message: fmt.Sprintf("could not serialize ledger entry data for ledger entry %v", ledgerKeyAndEntry.Entry),
+				ledgerEntryResults = append(ledgerEntryResults, LedgerEntryResult{
+					KeyJSON:            keyJs,
+					DataJSON:           entryJs,
+					LastModifiedLedger: uint32(ledgerKeyAndEntry.Entry.LastModifiedLedgerSeq),
+					LiveUntilLedgerSeq: ledgerKeyAndEntry.LiveUntilLedgerSeq,
+				})
+
+			default:
+				keyXDR, err := xdr.MarshalBase64(ledgerKeyAndEntry.Key)
+				if err != nil {
+					logger.WithError(err).WithField("request", request).
```

Comment:
nit: we seem to be inconsistent about whether we log serialization errors. In this endpoint we log marshling errors only for bas64 but not for json. in other endpoints we never log  serialization errors

Reply:
Yeah I agree with you but didn't want to modify this code (it's unchanged). Removed in abeffe9 :+1: